.Lib9c.Tests/Action/Scenario/WorldUnlockScenarioTest.cs

```diff
@@ -0,0 +1,268 @@
+            Assert.True(_tableSheets.StageSheet.ContainsKey(stageIdToClear));
+            Assert.True(_tableSheets.WorldSheet.ContainsKey(worldIdToUnlock));
+            Assert.False(_tableSheets.WorldUnlockSheet.TryGetUnlockedInformation(
+                worldIdToClear,
+                stageIdToClear,
+                out _));
+
+            var avatarState = _initialState.GetAvatarState(_avatarAddress);
+            avatarState.level = avatarLevel;
+
+            avatarState.worldInformation = new WorldInformation(0, _tableSheets.WorldSheet, stageIdToClear);
+            Assert.True(avatarState.worldInformation.IsWorldUnlocked(worldIdToClear));
+            Assert.False(avatarState.worldInformation.IsWorldUnlocked(worldIdToUnlock));
+
+            var nextState = _initialState.SetState(_avatarAddress, avatarState.Serialize());
+            var hackAndSlash = new HackAndSlash3
+            {
+                worldId = worldIdToClear,
+                stageId = stageIdToClear,
+                avatarAddress = _avatarAddress,
+                costumes = new List<int>(),
+                equipments = new List<Guid>(),
+                foods = new List<Guid>(),
+                WeeklyArenaAddress = _weeklyArenaState.address,
+                RankingMapAddress = _rankingMapAddress,
+            };
+            nextState = hackAndSlash.Execute(new ActionContext
+            {
+                PreviousStates = nextState,
+                Signer = _agentAddress,
+                Random = new TestRandom(),
+                Rehearsal = false,
+            });
+            Assert.True(hackAndSlash.Result.IsClear);
+
+            var mimisbrunnrBattle = new MimisbrunnrBattle()
+            {
+                worldId = worldIdToUnlock,
+                stageId = stageIdToUnlock,
+                avatarAddress = _avatarAddress,
+                costumes = new List<Guid>(),
+                equipments = new List<Guid>(),
+                foods = new List<Guid>(),
+                WeeklyArenaAddress = _weeklyArenaState.address,
+                RankingMapAddress = _rankingMapAddress,
+            };
+
+            Assert.Null(mimisbrunnrBattle.Result);
+
+            Assert.Throws<InvalidWorldException>(() =>
+            {
+                mimisbrunnrBattle.Execute(new ActionContext
+                {
+                    PreviousStates = nextState,
+                    Signer = _agentAddress,
+                    Random = new TestRandom(),
+                    Rehearsal = false,
+                });
+            });
+
+            avatarState = nextState.GetAvatarState(_avatarAddress);
+            Assert.True(avatarState.worldInformation.IsStageCleared(stageIdToClear));
+            Assert.False(avatarState.worldInformation.IsWorldUnlocked(worldIdToUnlock));
+
+            var tableCsv = _initialState.GetSheetCsv<WorldUnlockSheet>();
+            var newTable = new StringBuilder(tableCsv).Replace("4,2,100,10001", "4,2,80,10001").ToString();
```

Comment:
We need to subtract the hardcoded part of this line as a factor. For example..

```suggestion
            var newTable = new StringBuilder(tableCsv).Replace(targetRowStringBefore, targetRowStringAfter).ToString();
```